azurerm/internal/services/springcloud/spring_cloud_app_redis_binding_resource_test.go

```diff
@@ -0,0 +1,159 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/springcloud/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+type SpringCloudAppRedisBindingResource struct {
+}
+
+func TestAccSpringCloudAppRedisBinding_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_spring_cloud_app_redis_binding", "test")
+	r := SpringCloudAppRedisBindingResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("redis_access_key"),
+	})
+}
+
+func TestAccSpringCloudAppRedisBinding_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_spring_cloud_app_redis_binding", "test")
+	r := SpringCloudAppRedisBindingResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccSpringCloudAppRedisBinding_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_spring_cloud_app_redis_binding", "test")
+	r := SpringCloudAppRedisBindingResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("redis_access_key"),
+		{
+			Config: r.update(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("redis_access_key"),
+	})
+}
+
+func (t SpringCloudAppRedisBindingResource) Exists(ctx context.Context, clients *clients.Client, state *terraform.InstanceState) (*bool, error) {
+	id, err := parse.SpringCloudAppBindingID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.AppPlatform.BindingsClient.Get(ctx, id.ResourceGroup, id.SpringName, id.AppName, id.BindingName)
+	if err != nil {
+		return nil, fmt.Errorf("reading %s: %+v", id, err)
+	}
+
+	return utils.Bool(resp.Properties != nil), nil
+}
+
+func (r SpringCloudAppRedisBindingResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_spring_cloud_app_redis_binding" "test" {
+  name                = "acctestscarb-%d"
+  spring_cloud_app_id = azurerm_spring_cloud_app.test.id
+  redis_cache_id      = azurerm_redis_cache.test.id
+  redis_access_key    = azurerm_redis_cache.test.primary_access_key
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r SpringCloudAppRedisBindingResource) update(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_spring_cloud_app_redis_binding" "test" {
+  name                = "acctestscarb-%d"
```

Comment:
are capitals allowed here? if so we should use some

Reply:
only lower case letters, numbers and hythens are allowed. In the validation func, I have added the rule